Add plugin: Notebook Navigator

johansan opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/johansan/notebook-navigator

## Release Checklist
- [X] I have tested the plugin on
  - [X]  Windows
  - [X]  macOS
  - [ ]  Linux
  - [X]  Android _(if applicable)_
  - [X]  iOS _(if applicable)_
- [X] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [X] `main.js`
  - [X] `manifest.json`
  - [X] `styles.css` _(optional)_
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [X] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] My README.md describes the plugin's purpose and provides clear usage instructions.
- [X] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [X] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [X] I have added a license in the LICENSE file.
- [X] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[this.ribbonIconEl?.remove();](https://github.com/johansan/notebook-navigator/blob/6a2d3a3a6255429fa9ac4b1d1b41c623fca0e09a/src/main.ts#L483-L484)
This should not be necessary.

[localStorage.remove(key);](https://github.com/johansan/notebook-navigator/blob/6a2d3a3a6255429fa9ac4b1d1b41c623fca0e09a/src/main.ts#L494), [const item = window.localStorage.getItem(key);](https://github.com/johansan/notebook-navigator/blob/6a2d3a3a6255429fa9ac4b1d1b41c623fca0e09a/src/utils/localStorage.ts#L32), [window.localStorage.setItem(key, stringified);](https://github.com/johansan/notebook-navigator/blob/6a2d3a3a6255429fa9ac4b1d1b41c623fca0e09a/src/utils/localStorage.ts#L52), [window.localStorage.removeItem(key);](https://github.com/johansan/notebook-navigator/blob/6a2d3a3a6255429fa9ac4b1d1b41c623fca0e09a/src/utils/localStorage.ts#L67)
Prefer to use [App.saveLocalStorage](https://docs.obsidian.md/Reference/TypeScript+API/App/saveLocalStorage) and [App.loadLocalStorage](https://docs.obsidian.md/Reference/TypeScript+API/App/loadLocalStorage) to save and load vault-specific data to local storage. Using `localStorage.setItem` and `localStorage.getItem` will share that data across vaults on the same device.

[* Refreshes the ribbon icon based on current view state](https://github.com/johansan/notebook-navigator/blob/6a2d3a3a6255429fa9ac4b1d1b41c623fca0e09a/src/main.ts#L611-L628)
The ribbon icons should not be dependent on some state, that will screw up any custom ribbon order a user might have set.

[element.toggleClass('nn-setting-hidden', !show);](https://github.com/johansan/notebook-navigator/blob/6a2d3a3a6255429fa9ac4b1d1b41c623fca0e09a/src/settings.ts#L266)
This can be replaced with `element.toggle(!show)`

[export function isTFile(obj: unknown): obj is TFile {](https://github.com/johansan/notebook-navigator/blob/6a2d3a3a6255429fa9ac4b1d1b41c623fca0e09a/src/utils/typeGuards.ts#L25-L27)[export function isTFolder(obj: unknown): obj is TFolder {](https://github.com/johansan/notebook-navigator/blob/6a2d3a3a6255429fa9ac4b1d1b41c623fca0e09a/src/utils/typeGuards.ts#L32-L42), [export function isFileView(view: View): view is FileView {](https://github.com/johansan/notebook-navigator/blob/6a2d3a3a6255429fa9ac4b1d1b41c623fca0e09a/src/utils/typeGuards.ts#L47-L49)
Why can't you use a `instanceof` check?

johansan commented:
Hi Johannes, let me know if there is anything else you need before approving. I have so many people asking me every day now when it's going to be released. Have a nice day! ☀️

johansan commented:
Hi again @joethei 

I have now added the official https://github.com/obsidianmd/eslint-plugin to my build process. It found some issues and I have fixed them all in the latest release:

https://github.com/johansan/notebook-navigator/releases/tag/1.3.18

- Replaced some TFolder type assertions I had missed with proper instanceof TFolder checks
- Fixed command ID from "open-notebook-navigator" to "open-view" (no plugin ID in command)
- Replaced an inline style.marginLeft I had missed with CSS class

**Plugin is now 100% compatible with eslint-plugin-obsidianmd@0.1.4**

I am using ESLint v9 checking TypeScript, React, React Hooks, and Obsidian-specific rules

```
johan@Johans-MBP scripts % ./build.sh 
Running ESLint...

> notebook-navigator@1.3.18 lint
> eslint src --ext .ts,.tsx
✅ ESLint passed

Checking TypeScript types...
✅ TypeScript types are valid
Checking for unused imports...
✅ No unused imports found

Checking for dead code...
✅ No dead code found

Checking code formatting...
✅ Code formatting fixed (1 files updated, 145 unchanged)

Building notebook-navigator...

> notebook-navigator@1.3.18 build
> tsc -noEmit -skipLibCheck && node esbuild.config.mjs production

✅ Build completed successfully
Running local post-build script...
✅ Deployed to Obsidian vault

=== Build Summary ===

✅ Build successful
✅ No warnings
```

DrLeucine commented:
Congrats @joethei ! Looking forward installing this soon :)
